burn-book/src/building-blocks/tensor.md

```diff
@@ -0,0 +1,169 @@
+# Tensor
+
+As previously explained in the [model section](../basic-workflow/model.md), the Tensor struct has 3 generic arguments: the backend, the dimension number (rank), and the kind.
+
+```rust , ignore
+Tensor<B, D>           // Float tensor (default)
+Tensor<B, D, Float>    // Explicit float tensor
+Tensor<B, D, Int>      // Int tensor
+Tensor<B, D, Bool>     // Bool tensor
+```
+
+Note that the specific element types used for `Float`, `Int`, and `Bool` tensors are defined by backend implementations.
+
+## Operations
+
+Almost all Burn operations take ownership of the input tensors.
+Therefore, reusing a tensor multiple times will necessitate cloning it.
+Don't worry, the tensor's buffer isn't copied, but a reference to it is increased.
+This makes it possible to determine exactly how many times a tensor is used, which is very convenient for reusing tensor buffers and improving performance.
+For that reason, we don't provide inplace operations.
+If a tensor is used only one time, inplace operations will always be used when available.
+
+Normally with PyTorch, inplace operations aren't supported during the backward pass, making them useful only for data preprocessing or inference-only model implementations.
+With Burn, you can focus more on _what_ the model should do, rather than on _how_ to do it.
+We take the responsibility of making your code run as fast as possible during training as well as inference.
```

Comment:
I mean explicit inplace operations will cause exception if you try to call backward with PyTorch. Since there is no explicit inplace operations with Burn, this problem doesn't exist.